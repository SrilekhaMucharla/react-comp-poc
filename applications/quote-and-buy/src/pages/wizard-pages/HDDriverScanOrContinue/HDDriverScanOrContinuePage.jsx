// react
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// Hastings
import {
    HDAlertRefactor, HDInfoCardRefactor, HDLabelRefactor
} from 'hastings-components';

import { useLocation } from 'react-router-dom';
// import styles from './HDDriverScanOrContinuePage.module.scss';
import { Col, Row, Container } from 'react-bootstrap';
import { AnalyticsHDButton as HDButton } from '../../../web-analytics';
import * as messages from './HDDriverScanOrContinuePage.messages';
import { setNavigation as setNavigationAction } from '../../../redux-thunk/actions';
import useAnotherDriver from '../__helpers__/useAnotherDriver';

// Resources
import discount from '../../../assets/images/wizard-images/discount.png';
import exclamationIcon from '../../../assets/images/icons/exclamation-icon.svg';
import tipIcon from '../../../assets/images/icons/tip_circle_purple.svg';
import useRouterPageContext from '../__helpers__/useRouterPageContext';

export const HDDriverScanOrContinuePage = (props) => {
    const {
        handleForward,
        handleSkip,
        setNavigation,
        multiCarFlag
    } = props;

    // eslint-disable-next-line no-unused-vars
    const [driverIndex, isAnotherDriver, isAnotherDriverMulti] = useAnotherDriver(useLocation());
    const routerPageContext = useRouterPageContext(useLocation());

    useEffect(() => {
        const wizardTooltip = (
            <HDInfoCardRefactor
                id="page-scan-or-continue-wizard-tooltip"
                image={tipIcon}
                paragraphs={[messages.wizardTooltip(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver)]} />
        );
        setNavigation({
            canSkip: false,
            canForward: false,
            showForward: false,
            showWizardTooltip: true,
            wizardTooltip: wizardTooltip
        });
    }, []);

    const handleScan = () => {
        handleForward({ routerPageContext: { errorMessage: null } });
    };

    const handleContinue = () => {
        handleSkip({ routerPageContext: { errorMessage: null } });
    };

    const discountTooltip = (<img src={discount} alt="discount" className="page-scan-or-continue__discount-img" />);

    return (
        <Container id="scan-or-continue" className="scan-or-continue">
            <Row>
                <Col>
                    <HDLabelRefactor
                        id="page-scan-or-continue-title"
                        className="page-scan-or-continue__title"
                        text={messages.scanOrContinueTitle(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver)}
                        Tag="h2" />
                    {isAnotherDriver && (
                        <HDInfoCardRefactor
                            id="page-scan-or-continue-other-driver-info"
                            className="page-scan-or-continue__other-driver-info"
                            image={exclamationIcon}
                            paragraphs={[messages.additionDriverInfo]}
                            size="thin" />
                    )}
                </Col>
            </Row>
            <Row className="page-scan-or-continue__subtitle">
                <Col className="align-self-center pr-0">
                    <HDLabelRefactor
                        id="page-scan-or-continue-subtitle"
                        className="page-scan-or-continue__subtitle__label"
                        text={messages.otherDriverLabel(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver)}
                        additionalContent={<h5 className="d-inline">{messages.optionalText(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver)}</h5>}
                        Tag="h4" />
                </Col>
                <Col xs="auto" className="pl-2">
                    {discountTooltip}
                </Col>
            </Row>
            <Row>
                <Col>
                    <HDLabelRefactor
                        id="page-scan-or-continue-disclaimer"
                        className="page-scan-or-continue__disclaimer text-small"
                        text={messages.disclaimer}
                        Tag="p" />
                    <HDInfoCardRefactor
                        id="page-scan-or-continue-driving-information"
                        className="page-scan-or-continue__driving-information"
                        image={exclamationIcon}
                        paragraphs={[messages.tooltipContentLine1(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver), messages.tooltipContentLine2(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver)]} />
                    {routerPageContext.errorMessage && <HDAlertRefactor message={routerPageContext.errorMessage} />}
                </Col>
            </Row>
            <Row className="margin-top-lg-desktop-md-mobile col-elem-same-height">
                <Col xs={12} sm={6}>
                    <HDButton
                        webAnalyticsEvent={{ event_action: messages.scanOrContinueTitle(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver) }}
                        id="page-scan-or-continue-button-scan-licence"
                        name="scan"
                        label={messages.scanLabel}
                        onClick={handleScan}
                        className="page-scan-or-continue__button-scan-licence btn-primary btn-block" />
                </Col>
                <Col xs={12} sm={6} className="mt-4 mt-sm-0">
                    <HDButton
                        webAnalyticsEvent={{ event_action: messages.scanOrContinueTitle(multiCarFlag ? isAnotherDriverMulti : isAnotherDriver) }}
                        id="page-scan-or-continue-button-continue"
                        name="continue"
                        label={messages.continueLabel}
                        onClick={handleContinue}
                        className="page-scan-or-continue__button-continue hd-btn-secondary btn-block" />
                </Col>
            </Row>
        </Container>
    );
};

const mapStateToProps = (state) => {
    return {
        multiCarFlag: state.wizardState.app.multiCarFlag
    };
};

const mapDispatchToProps = {
    setNavigation: setNavigationAction
};

HDDriverScanOrContinuePage.propTypes = {
    handleForward: PropTypes.func.isRequired,
    handleSkip: PropTypes.func.isRequired,
    setNavigation: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(HDDriverScanOrContinuePage);
