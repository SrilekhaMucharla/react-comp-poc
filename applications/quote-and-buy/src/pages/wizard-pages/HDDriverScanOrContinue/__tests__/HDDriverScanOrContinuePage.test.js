import React from 'react';
import { shallow } from 'enzyme';
import { HDDriverScanOrContinuePage } from '../HDDriverScanOrContinuePage';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({})
}));

describe('HDDriverScanPage', () => {
    const handleForward = jest.fn();
    const handleSkip = jest.fn();
    const wrapper = shallow((<HDDriverScanOrContinuePage handleSkip={handleSkip} handleForward={handleForward} setNavigation={jest.fn()} />));

    beforeEach(() => {

    });

    it('should render the component and match snapshot', async () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should navigate forward when clicking scan button', async () => {
        const scanButton = wrapper.find('[name="scan"]').at(0);
        scanButton.simulate('click');

        expect(handleForward).toBeCalled();
    });

    it('should skip when clicking continue button', async () => {
        const scanButton = wrapper.find('[name="continue"]').at(0);
        scanButton.simulate('click');

        expect(handleSkip).toBeCalled();
    });
});
