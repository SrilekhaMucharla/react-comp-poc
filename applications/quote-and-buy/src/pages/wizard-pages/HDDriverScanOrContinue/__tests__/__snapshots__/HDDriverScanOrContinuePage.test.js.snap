// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HDDriverScanPage should render the component and match snapshot 1`] = `
<Container
  className="scan-or-continue"
  fluid={false}
  id="scan-or-continue"
>
  <Row
    noGutters={false}
  >
    <Col>
      <HDLabelRefactor
        Tag="h2"
        adjustImagePosition={true}
        className="page-scan-or-continue__title"
        id="page-scan-or-continue-title"
        text="Want to speed things up?"
      />
    </Col>
  </Row>
  <Row
    className="page-scan-or-continue__subtitle"
    noGutters={false}
  >
    <Col
      className="align-self-center pr-0"
    >
      <HDLabelRefactor
        Tag="h4"
        additionalContent={
          <h5
            className="d-inline"
          />
        }
        adjustImagePosition={true}
        className="page-scan-or-continue__subtitle__label"
        id="page-scan-or-continue-subtitle"
        text="You'll get a faster quote if you scan your driving licence..."
      />
    </Col>
    <Col
      className="pl-2"
      xs="auto"
    >
      <img
        alt="discount"
        className="page-scan-or-continue__discount-img"
        src="discount.png"
      />
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <HDLabelRefactor
        Tag="p"
        adjustImagePosition={true}
        className="page-scan-or-continue__disclaimer text-small"
        id="page-scan-or-continue-disclaimer"
        text="*Based on online quoted premiums from June to August 2021"
      />
      <HDInfoCardRefactor
        className="page-scan-or-continue__driving-information"
        id="page-scan-or-continue-driving-information"
        image="exclamation-icon.svg"
        isRight={false}
        notes={null}
        paragraphs={
          Array [
            "We'll be able to access your driving information from the DVLA, so we can calculate any discounts you qualify for based on your driving record.",
            "Your licence must be from England, Wales or Scotland as DVLA data isn't available for licences from Northern Ireland, the Isle of Man or the Channel Islands.",
          ]
        }
        size={null}
        theme={null}
        title={null}
      />
    </Col>
  </Row>
  <Row
    className="margin-top-lg-desktop-md-mobile col-elem-same-height"
    noGutters={false}
  >
    <Col
      sm={6}
      xs={12}
    >
      <ForwardRef
        className="page-scan-or-continue__button-scan-licence btn-primary btn-block"
        id="page-scan-or-continue-button-scan-licence"
        label="Scan driving licence"
        name="scan"
        onBeforeOpen={null}
        onBlur={null}
        onCancel={null}
        onChange={null}
        onClick={[Function]}
        onConfirm={null}
        onDelete={null}
        onEdit={null}
        onFocus={null}
        onSelect={null}
        webAnalyticsEvent={
          Object {
            "event_action": "Want to speed things up?",
          }
        }
      />
    </Col>
    <Col
      className="mt-4 mt-sm-0"
      sm={6}
      xs={12}
    >
      <ForwardRef
        className="page-scan-or-continue__button-continue hd-btn-secondary btn-block"
        id="page-scan-or-continue-button-continue"
        label="Continue (without scanning)"
        name="continue"
        onBeforeOpen={null}
        onBlur={null}
        onCancel={null}
        onChange={null}
        onClick={[Function]}
        onConfirm={null}
        onDelete={null}
        onEdit={null}
        onFocus={null}
        onSelect={null}
        webAnalyticsEvent={
          Object {
            "event_action": "Want to speed things up?",
          }
        }
      />
    </Col>
  </Row>
</Container>
`;
